build/circle-test.sh

```diff
@@ -24,7 +24,8 @@ test ${PIPESTATUS[0]} -eq 0
 # 3. Run "make test".
 match='^panic|^[Gg]oroutine \d+|(read|write) by.*goroutine|DATA RACE'
 echo "make test"
-time ${builder} make GITHOOKS= test \
+time ${builder} make test GITHOOKS= \
+     CPUS=2
```

Comment:
Do we really need to change this? It should be harmless to run with GOMAXPROCS=4 even when the machine is only two cpus. If we do want to change this it would be nice to make it dynamic; the most standard way to do this from the command line appears to be `getconf _NPROCESSORS_ONLN` (which works on OSX and linux).